controllers/dspipeline_controller.go

```diff
@@ -294,6 +305,22 @@ func (r *DSPAReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 			return ctrl.Result{}, err
 		}
 
+		if dspa.Spec.APIServer.PipelineStorage == "kubernetes" {
+			err = r.ReconcileWebhook(ctx, params)
+			if err != nil {
+				dspaStatus.SetWebhookNotReady(err, config.FailingToDeploy)
+				return ctrl.Result{}, err
+			}
+			ready, reason := r.checkWebhookStatus(ctx, params)
+			if ready {
+				dspaStatus.SetWebhookReady()
+			} else {
+				dspaStatus.SetWebhookNotReady(nil, reason)
```

Comment:
This will cause the DSPA to be reconciled again soon and check on the webhook status again.
```suggestion
				dspaStatus.SetWebhookNotReady(nil, reason)
				dspaPrereqsReady = false
```